src/v2/Apps/Artist/Components/ArtistTopWorksRail/ArtistTopWorksRail.tsx

```diff
@@ -0,0 +1,150 @@
+      const { top } = getElementPosition($element)
+      window.scrollTo({
+        top: top,
+      })
+    }
+  }
+
+  const getElementPosition = $element => {
+    const rect = $element.getBoundingClientRect()
+    return {
+      left: rect.left + window.scrollX,
+      top: rect.top + window.scrollY,
+    }
+  }
+
+  return artworks.length > 0 ? (
+    <Flex mb="75px" flexDirection="column">
+      <Flex my={1} justifyContent="space-between">
+        <Sans size="4" color="black100">
+          Top Works
+        </Sans>
+        <Media at="xs">
+          {onOverviewTab ? (
+            <StyledLink
+              onClick={e => handleViewWorksClick(e)}
+              to={`/artist/${topRailArtworks.slug}/works-for-sale`}
+            >
+              <Sans size="4" color="black60">
+                View all
+              </Sans>
+            </StyledLink>
+          ) : (
+            <Box
+              onClick={e => scrollToArtworkView(e)}
+              style={{ cursor: "pointer" }}
+            >
+              <Sans size="4" color="black60">
+                View all
+              </Sans>
+            </Box>
+          )}
+        </Media>
+        <Media greaterThan="xs">
+          {onOverviewTab ? (
+            <StyledLink
+              onClick={e => handleViewWorksClick(e)}
+              data-test="link-to-works-for-sale"
+              to={`/artist/${topRailArtworks.slug}/works-for-sale`}
+            >
+              <Sans size="4" color="black60">
+                View all works
+              </Sans>
+            </StyledLink>
+          ) : (
+            <Box
+              onClick={e => scrollToArtworkView(e)}
+              style={{ cursor: "pointer" }}
+            >
+              <Sans size="4" color="black60">
+                View all works
+              </Sans>
+            </Box>
+          )}
+        </Media>
+      </Flex>
+      <Carousel
+        data={artworks}
+        height={carouselHeight}
+        options={{
+          pageDots: false,
+        }}
+        render={({ node }) => {
+          const { image } = node
+
+          return (
+            <Box height={376} mb={3} width="auto">
+              <FillwidthItem
+                artwork={node}
+                hidePartnerName
+                hideArtistName
+                useLighterFont
+                targetHeight={image?.resized?.height}
+                imageHeight={image?.resized?.height}
+                width={image?.resized?.height * image?.imageAspectRatio}
+                margin={5}
+                showMetadata
+                showExtended={false}
+              />
+            </Box>
+          )
+        }}
+      />
+    </Flex>
+  ) : null
+}
+
+export const ArtistTopWorksRailFragmentContainer = createFragmentContainer(
+  ArtistTopWorksRail,
+  {
+    topRailArtworks: graphql`
+      fragment ArtistTopWorksRail_topRailArtworks on Artist {
+        slug
+        images: artworksConnection(first: 10, sort: ICONICITY_DESC) {
```

Comment:
I would maybe keep the naming as-is and remove the alias (so it stays as `artworksConnection`). It looks like the edges gets renamed to `artworks` anyway in https://github.com/artsy/force/pull/5647/files#diff-8e2b33c841ad707ecacb196c63a21890R20

Reply:
Agree with this! The old `images` naming seems outdated now that we're returning proper `Artwork`s here.

Reply:
I'm getting this error, if I don't alias:

```
ERROR:
Expected all fields on the same parent with the name or alias 'artworksConnection' to have the same name and arguments.

Source: src/v2/Apps/Artist/Components/ArtistTopWorksRail/ArtistTopWorksRail.tsx:124:9
123:         slug
124:         artworksConnection(first: 10, sort: ICONICITY_DESC) {
             ^
125:           edges {

Source: src/v2/Apps/Artist/Routes/Overview/Components/WorksForSaleRail.tsx:107:9
106:       fragment WorksForSaleRail_artist on Artist {
107:         artworksConnection(first: 20, sort: AVAILABILITY_ASC) {
             ^
108:           edges {
```

Reply:
Ah! I'd just give it a different alias then. `topWorksArtworks: artworksConnection` or something like that!